app/src/main/java/org/mozilla/fenix/components/appstate/AppState.kt

```diff
@@ -4,14 +4,17 @@
 
 package org.mozilla.fenix.components.appstate
 
+import mozilla.components.lib.crash.Crash
 import mozilla.components.lib.state.State
 
 /**
  * Value type that represents the state of the tabs tray.
  *
  * @property inactiveTabsExpanded A flag to know if the Inactive Tabs section of the Tabs Tray
  * should be expanded when the tray is opened.
+ * @property nonFatalCrashes List of non-fatal crashes that allow the app to continue being used.
  */
 data class AppState(
-    val inactiveTabsExpanded: Boolean = false
+    val inactiveTabsExpanded: Boolean = false,
+    val nonFatalCrashes: List<Crash> = emptyList()
```

Comment:
We receive the content crash(es) through an intent and keep them in memory as pending until the user decides to report them.
First looked into keeping them in BrowserState and completely avoid depending on the crash intent but for that I needed to map a crash to a particular tabId, something not possible atm.

Reply:
Since we're only storing native non-fatal crashes here, should this be `<Crash.UncaughtExceptionCrash>` so it's clear?

Reply:
Thanks! This seems like a nice improvement.